File: go-app/internal/models/sale.go
package models

import (
	"database/sql"
	"time"

	"gorm.io/gorm"
)

type Sale struct {
	ID            uint           `gorm:"primaryKey" json:"id"`
	EbayOrderID   string         `gorm:"column:ebay_order_id;unique" json:"ebay_order_id"`
	SaleDate      time.Time      `gorm:"column:sale_date" json:"sale_date"`
	Title         string         `json:"title"`
	BuyerUsername string         `gorm:"column:buyer_username" json:"buyer_username"`
	GrossAmount   float64        `gorm:"column:gross_amount" json:"gross_amount"`
	EbayFees      float64        `gorm:"column:ebay_fees" json:"ebay_fees"`
	ShippingCost  float64        `gorm:"column:shipping_cost" json:"shipping_cost"`
	NetAmount     float64        `gorm:"column:net_amount" json:"net_amount"`
	ImageURL      string         `gorm:"column:image_url" json:"image_url"`
	OrderStatus   string         `gorm:"column:order_status" json:"order_status"`
	Origin        string         `gorm:"column:origin;default:EBAY" json:"origin"`
	Status        string         `gorm:"column:status;default:STAGED" json:"status"`
	AttributedTo  string         `gorm:"column:attributed_to" json:"attributed_to"`
	Notes         sql.NullString `gorm:"type:text" json:"notes"`
	Items         []TrackedItem  `gorm:"foreignKey:SaleID" json:"items,omitempty"`
	CreatedAt     time.Time      `json:"created_at"`
	UpdatedAt     time.Time      `json:"updated_at"`
	DeletedAt     gorm.DeletedAt `gorm:"index" json:"-"`
}

File: src/main/java/com/collectingwithzak/entity/Sale.java
package com.collectingwithzak.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.SQLRestriction;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "sales")
@Getter
@Setter
@SQLDelete(sql = "UPDATE sales SET deleted_at = NOW() WHERE id = ?")
@SQLRestriction("deleted_at IS NULL")
public class Sale extends BaseEntity {

    @Column(name = "ebay_order_id", unique = true)
    private String ebayOrderId;

    @Column(name = "sale_date")
    private LocalDate saleDate;

    private String title;

    @Column(name = "buyer_username")
    private String buyerUsername;

    @Column(name = "gross_amount", columnDefinition = "numeric(10,2)")
    private double grossAmount;

    @Column(name = "ebay_fees", columnDefinition = "numeric(10,2)")
    private double ebayFees;

    @Column(name = "shipping_cost", columnDefinition = "numeric(10,2)")
    private double shippingCost;

    @Column(name = "net_amount", columnDefinition = "numeric(10,2)")
    private double netAmount;

    @Column(name = "image_url")
    private String imageUrl;

    @Column(name = "order_status")
    private String orderStatus;

    private String origin = "EBAY";

    private String status = "STAGED";

    @Column(name = "attributed_to")
    private String attributedTo;

    private String notes;

    @OneToMany(mappedBy = "sale", fetch = FetchType.LAZY)
    private List<TrackedItem> items = new ArrayList<>();
}
